### DISCUSSION TITLE
script installer, name convention and name resolution

### DISCUSSION THREAD

**Discussion by timreichen** [denoland/deno#9057] [Ideas]
Summary of points of discussion about script installer, name convention and name resolution from discord

1. Name convention for _installable_ files
**current status**: no convention
**suggestions**
   - install.ts and *_install.ts like in test.ts and *_test.ts
   - cli.ts

2. resolution of `deno install` when `--name` is not set
**current status**: The executable name is inferred by default:
  Attempt to take the file stem of the URL path. The above example would become 'file_server'.
  If the file stem is something generic like 'main', 'mod', 'index' or 'cli', and the path has no parent, take the file name of the parent path. Otherwise settle with the generic name.
  If the resulting name has an '@...' suffix, strip it.
**suggestions**:
   - always resolve to module name (and no special treating of 'main', 'mod', 'index' and 'cli')
   - mandatory `--name`

3. mandatory `--name` in `deno install`
**current status**: not mandatory
**suggestion**: should be mandatory

**Comment by wperron** [comment]
Thanks @nayeemrmn for the original issue on this #5004

**Comment by mjrlowe** [comment]
[Following on from our discussion on Discord] Regarding point 1, personally I am in favour of `cli.ts` being the community convention, since they don't have to be installed to run as a CLI tool (but installable TS/JS files will always function as a CLI tool when run).
